src/spikeinterface/widgets/unit_summary.py

```diff
@@ -136,37 +140,41 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
             ax1.set_xlabel(None)
             ax1.set_ylabel(None)
 
-        ax2 = fig.add_subplot(gs[:2, 1])
+        ax2 = fig.add_subplot(gs[:2, col_counter])
         w = UnitWaveformsWidget(
             sorting_analyzer,
             unit_ids=[unit_id],
             unit_colors=unit_colors,
             plot_templates=True,
+            plot_waveforms=sorting_analyzer.has_extension("waveforms"),
             same_axis=True,
             plot_legend=False,
             sparsity=sparsity,
             backend="matplotlib",
             ax=ax2,
             **unitwaveformswidget_kwargs,
         )
+        col_counter = col_counter + 1
 
         ax2.set_title(None)
 
-        ax3 = fig.add_subplot(gs[:2, 2])
-        UnitWaveformDensityMapWidget(
-            sorting_analyzer,
-            unit_ids=[unit_id],
-            unit_colors=unit_colors,
-            use_max_channel=True,
-            same_axis=False,
-            backend="matplotlib",
-            ax=ax3,
-            **unitwaveformdensitymapwidget_kwargs,
-        )
-        ax3.set_ylabel(None)
+        if sorting_analyzer.has_extension("waveforms"):
+            ax3 = fig.add_subplot(gs[:2, col_counter])
+            UnitWaveformDensityMapWidget(
+                sorting_analyzer,
+                unit_ids=[unit_id],
+                unit_colors=unit_colors,
+                use_max_channel=True,
+                same_axis=False,
+                backend="matplotlib",
+                ax=ax3,
+                **unitwaveformdensitymapwidget_kwargs,
+            )
+            ax3.set_ylabel(None)
+            col_counter = col_counter + 1
```

Comment:
these could be `col_counter += 1`

Reply:
I know that `c++` isn't in python, and this somehow merged in my mind with the `+=` operator. Omg I've been coding in python for 9 months now... thanks Joe!